www/FrequentlyAskedQuestions.md

```diff
@@ -240,3 +240,47 @@ a combination of TypeMapper and FunctionMapper (see
 `com.sun.jna.win32.W32APIOptions.DEFAULT_OPTIONS`) so that you can leave off the 
 “-A” or “-W” suffix (you never need to use both simultaneously) and use 
 “String” rather than explicit “WString”.
+
+Does JNA publish a module descriptor (module-info.java) to support the Java Module System (JPMS)?
+-------------------------------------------------------------------------------------------------------------------------------------------
+
+Since version 5.7.0, JNA publishes an additional JAR with a `module-info` class alongside
+the main project JAR, using the `jpms` qualifier, e.g. `jna-5.7.0-jpms.jar`. For a
+Maven build, use the following dependency statement:
+```
+<dependency>
+  <groupId>net.java.dev.jna</groupId>
+  <artifactId>jna</artifactId>
+  <version>5.7.0</version>
+  <classifier>jpms</classifier>
+</dependency>
+```
+and include `requires com.sun.jna;` in your module descriptor in your `module-info.java` file.
+
+If you use the `jna-platform` artifact, it depends transitively on `jna` and requires
+special handling to exclude the non-modular JAR:
+```
+<dependency>
+  <groupId>net.java.dev.jna</groupId>
+  <artifactId>jna-platform</artifactId>
+  <version>5.7.0</version>
+  <classifier>jpms</classifier>
+  <exclusions>
+    <exclusion>
+      <groupId>net.java.dev.jna</groupId>
+      <artifactId>jna</artifactId>
+    </exclusion>
+  </exclusions>
+</dependency>
+```
+and include `requires com.sun.jna.platform;` in your module descriptor.
+
+If you have a library that may be consumed by downstream users, consider making these managed dependencies.
+
+In addition to adding the `requires` directive, note that several JNA classes
+make use of reflection, which is disabled by the modlue system's strong encapsulation.
+It may be necessary to make packages which include subclasses of JNA's classes
+(such as `Structure` and `PointerType` among others) accessible via reflection
+to the `com.sun.jna` module using either an `open`, `opens`, or
+`opens ... to` directive, or an `exports` or `exports ... to` directive,
+depending on the particular application and level of access desired.
```

Comment:
I would probably say "level of access required" here?  Possibly a note that to replicate the access that JNA has when used on the classpath (eg. when migrating an existing project or building something designed for both) use `opens`.

Reply:
Updated.